### ISSUE TITLE
Optional dependencies are getting built as required dependencies

### ISSUE THREAD

**Issue by monkeez** [sdispater/poetry#24]
Given this `pyproject.toml` file:
```
[tool.poetry]
name = "example"
version = "0.1.0"
authors = [ "My Name <example@gmail.com>",]
description = "Example package"

[tool.poetry.dependencies]
python = "^2.7"
pytest = { version = "^3.4", optional = true }

[tool.poetry.dev-dependencies]

[tool.poetry.extras]
test = ["pytest"]
```
When I run `poetry build`, if I then go and inspect the sdist that was built, the `setup.py` file looks like this:
```python
# -*- coding: utf-8 -*-
from distutils.core import setup

packages = \
['example']

package_data = \
{'': ['*']}

install_requires = \
['pytest (>=3.4.0.0,<4.0.0.0)']

setup_kwargs = {
    'name': 'example',
    'version': '0.1.0',
    'description': 'Example package',
    'long_description': None,
    'author': 'My Name',
    'author_email': 'example@gmail.com',
    'url': None,
    'packages': packages,
    'package_data': package_data,
    'install_requires': install_requires,
    'python_requires': '>= 2.7.0.0, < 3.0.0.0',
}


setup(**setup_kwargs)
```
As you can see `pytest` has been added to `install_requires` and not `extras_require`.

**Comment by sdispater** [comment]
This is definitely a bug. I will look at it and get back to you.

**Comment by sdispater** [comment]
This is fixed in the latest release (`0.6.4`)

**Comment by monkeez** [comment]
Thanks for all the work you've put into this project!

This is now working for `sdist` but I noticed that optional dependencies are still being set as required dependencies in the `wheel` build. This is the `METADATA` file from the same project as before:

```
Metadata-Version: 2.1
Name: example
Version: 0.1.0
Summary: Example package
Home-page: UNKNOWN
License: UNKOWN
Author: My Name
Author-email: example@gmail.com
Requires-Python: >= 2.7.0.0, < 3.0.0.0
Classifier: Programming Language :: Python :: 2
Classifier: Programming Language :: Python :: 2.7
Requires-Dist: pytest (>=3.4.0.0,<4.0.0.0)
```
Where this is the expected result:
```
Metadata-Version: 2.1
Name: example
Version: 0.1.0
Summary: Example package
Home-page: UNKNOWN
License: UNKOWN
Author: My Name
Author-email: example@gmail.com
Requires-Python: >= 2.7.0.0, < 3.0.0.0
Classifier: Programming Language :: Python :: 2
Classifier: Programming Language :: Python :: 2.7
Provides-Extra: test
Requires-Dist: pytest (>=3.4.0.0,<4.0.0.0); extra == 'test'
```
Thanks again.

**Comment by sdispater** [comment]
You are right. I somehow missed it.

I will fix it and get back to you.

**Comment by sdispater** [comment]
This should be fixed in release 0.6.5.

I will leave the issue open for now. Feel free to close it if you can confirm it now works as expected.

**Comment by monkeez** [comment]
Thanks! Working as intended on my end now.
